### PR TITLE
codeintel: Do not alert on limit errors

### PR DESCRIPTION
**Pull request by efritz**
Put error filters around inference errors.

## Test plan

Tested locally, saw less blather.

### CODE REVIEW

File: enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func NewScheduler(
 			Metrics:           redMetrics,
 			ErrorFilter: func(err error) observation.ErrorFilterBehaviour {
 				if errors.As(err, &inference.LimitError{}) {
-					return observation.EmitForDefault.Without(observation.EmitForMetrics)
+					return observation.EmitForNone
```
Comment (Strum355) [inline enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go:76]: not even logs? :eyes:

Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go:76]: Error log, nah, it's like a 400-level error.

Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go:76]: Or am I misunderstanding this filter?

Comment (Strum355) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go:76]: 400-level?

Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go:76]: User issue (missing resource, bad auth) not server issue (500 level, actual error to report)

**Review by Numbers88s — APPROVED** [review]

File: enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go (hunk 1 above)
Comment (Strum355) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go:76]: Is it fine to have it silently not do anything in that case? whats the expected method for admins to see why nothing is happening for example (scoping out the thought process here)

Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_scheduler.go:76]: We're also exposing it via the UI here: https://github.com/sourcegraph/sourcegraph/pull/47748.

**Pull request merged**